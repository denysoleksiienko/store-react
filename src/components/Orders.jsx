import React from 'react';
import {
  SummaryTitle,
  SummaryWrap,
  SummaryList,
  SummaryCost,
  SummaryItem,
  SummaryItemInfo,
  SummaryTotal,
  ProductInfo,
} from 'styled';

export const Orders = ({ orders, fetchProducts }) => {
  React.useEffect(() => {
    fetchProducts();
  }, [fetchProducts]);

  return (
    <>
      <SummaryTitle>Order Summary</SummaryTitle>
      <SummaryWrap>
        <SummaryList>
          {orders.map(({ id, img, title, color, qty, price }) => (
            <SummaryItem key={id}>
              <SummaryItemInfo>
                <img src={img} alt={title} />

                <ProductInfo>
                  <span>{title}</span>
                  <span>{color}</span>
                  <span>{qty}</span>
                </ProductInfo>

                <span>${price}</span>
              </SummaryItemInfo>
            </SummaryItem>
          ))}
        </SummaryList>

        <SummaryCost>
          <p>Subtotal</p>
          <p>Shipping</p>
          <p>Taxes</p>
          <hr />
          <SummaryTotal>Total</SummaryTotal>
        </SummaryCost>
      </SummaryWrap>
    </>
  );
};
