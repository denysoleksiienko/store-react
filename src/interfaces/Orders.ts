import { IProducts } from './Products';

export interface IOrders {
  orders: IProducts[];
  fetchProducts: Function;
}

export interface IOrderItem {
  img: string;
  title: string;
  color: string;
  qty: number;
  price: number;
}

export interface IOrderCost {
  subtotal: number;
}
export interface IFinishOrder {
  readonly user: {
    email: string;
  };
}
