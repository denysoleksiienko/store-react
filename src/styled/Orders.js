import styled from 'styled-components';
import { COLORS } from '../constants/colors';

export const SummaryTitle = styled.h5`
  font-size: 1.2em;
  color: ${COLORS.orchid};
  margin: 12px;
`;

export const SummaryWrap = styled.div`
  margin-left: 12px;
`;

export const SummaryList = styled.div`
  padding: 0;
`;

export const SummaryCost = styled.div`
  flex-flow: column;
  color: ${COLORS.gray};
`;

export const SummaryItem = styled.div`
  display: flex;
  justify-content: space-between;
  padding-bottom: 10px;
  margin-bottom: 10px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
`;

export const ProductInfo = styled.div`
  display: flex;
  flex-direction: column;
  width: 65%;
  & :first-child {
    color: ${COLORS.gray};
  }
`;

export const SummaryTotal = styled.span`
  font-weight: bold;
  color: ${COLORS.indigo};
`;

export const SummaryCostInner = styled.div`
  display: flex;
  justify-content: space-between;
`;
