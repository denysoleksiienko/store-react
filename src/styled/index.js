export { RowWrap, ColOrders } from './App';
export { BreadcrumbWrap, NavLinkStyled } from './Breadcrumbs';
export { StyledButton } from './Button';
export { SecureInfo, Icon, Span } from './Payment';
export { InputSpan } from './ContactFields';
export { OrderWrap } from './FinishOrders';
export {
  SummaryTitle,
  SummaryWrap,
  SummaryList,
  SummaryItem,
  ProductInfo,
  SummaryCost,
  SummaryTotal,
  SummaryCostInner,
} from './Orders';

export { Title } from './shared/Title';
export { Label } from './shared/Label';
export { InputsWrap } from './shared/InputsWrap';
export { FormGroupInner } from './shared/FormGroupInner';
