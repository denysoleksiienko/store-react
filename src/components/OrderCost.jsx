import { SummaryCost, SummaryTotal, SummaryCostInner } from 'styled';

export const OrderCost = ({ subtotal }) => (
  <SummaryCost>
    <SummaryCostInner>
      <span>Subtotal</span>
      <span>${subtotal}</span>
    </SummaryCostInner>
    <SummaryCostInner>
      <span>Shipping</span>
      <span>Free</span>
    </SummaryCostInner>
    <SummaryCostInner>
      <span>Taxes</span>
      <span>$10</span>
    </SummaryCostInner>
    <hr />

    <SummaryCostInner>
      <SummaryTotal>Total</SummaryTotal>
      <SummaryTotal>${subtotal + 10}</SummaryTotal>
    </SummaryCostInner>
  </SummaryCost>
);
