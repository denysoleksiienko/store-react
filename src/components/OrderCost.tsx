import React from 'react';
import { IOrderCost } from 'interfaces/Orders';
import { SummaryCost, SummaryTotal, SummaryCostInner } from 'styled';

export const OrderCost: React.FC<IOrderCost> = ({ subtotal }) => (
  <SummaryCost>
    <SummaryCostInner>
      <span>Subtotal</span>
      <span>${subtotal}</span>
    </SummaryCostInner>
    <SummaryCostInner>
      <span>Shipping</span>
      <span>Free</span>
    </SummaryCostInner>
    <SummaryCostInner>
      <span>Taxes</span>
      <span>$10</span>
    </SummaryCostInner>
    <hr />

    <SummaryCostInner>
      <SummaryTotal>Total</SummaryTotal>
      <SummaryTotal>${subtotal + 10}</SummaryTotal>
    </SummaryCostInner>
  </SummaryCost>
);
