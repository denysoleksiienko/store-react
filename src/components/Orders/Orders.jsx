import React from 'react';
import styled from 'styled-components';

// import './Orders.css';

export const Orders = () => {
  const [data, setData] = React.useState([
    {
      id: 1,
      name: 'The Chelsea Boot',
      color: 'black',
      qty: 1,
      price: '$235',
    },
    {
      id: 2,
      name: 'The Twill Snap Backpack',
      qty: 1,
      price: '$65',
    },
    {
      id: 3,
      name: 'The Twill Zip Tote',
      qty: 1,
      price: '$48',
    },
  ]);

  return (
    <>
      <Title>Order Summary</Title>
      <Summary>
        <List>
          {data.map(({ id, name, qty, price, color }) => (
            <Item key={id}>
              <span>
                {name} {price}
              </span>
              <span>{color}</span>
              <span>Quantity: {qty}</span>
            </Item>
          ))}
        </List>

        <Cost>
          <p>Subtotal</p>
          <p>Shipping</p>
          <p>Taxes</p>
          <hr />
          <Total>Total</Total>
        </Cost>
      </Summary>
    </>
  );
};

const Title = styled.h5`
  font-size: 1.2em;
  color: #9932cc;
  margin: 12px;
`;

const Summary = styled.div`
  margin-left: 12px;
`;

const Cost = styled.div`
  flex-flow: column;
`;

const List = styled.ul`
  list-style: none;
  padding: 0;
`;

const Item = styled.li`
  display: flex;
  flex-direction: column;
  padding-bottom: 10px;
  margin-bottom: 10px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
`;

const Total = styled.span`
  font-weight: bold;
  color: #310062;
`;
