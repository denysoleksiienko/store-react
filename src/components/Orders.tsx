import React from 'react';
import { OrderItem, OrderCost } from 'components';

import { SummaryTitle, SummaryWrap, SummaryList } from 'styled';

import { IOrders } from 'interfaces/Orders';
import { IProducts } from 'interfaces/Products';

export const Orders: React.FC<IOrders> = ({ orders, fetchProducts }) => {
  React.useEffect(() => {
    fetchProducts();
  }, [fetchProducts]);

  let subtotal: number = 0;

  return (
    <>
      <SummaryTitle>Order Summary</SummaryTitle>

      <SummaryWrap data-testid="summ-wrap">
        <SummaryList data-testid="summ-list">
          {orders.map(({ id, img, title, color, qty, price }: IProducts) => {
            subtotal += price;
            return <OrderItem key={id} img={img} title={title} color={color} qty={qty} price={price} />;
          })}
        </SummaryList>

        <OrderCost subtotal={subtotal} />
      </SummaryWrap>
    </>
  );
};
