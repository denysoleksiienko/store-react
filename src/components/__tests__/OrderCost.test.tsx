import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { OrderCost } from '../OrderCost';
import { IOrderCost } from 'interfaces/Orders';
import { SummaryCost } from 'styled';

const props = {
  subtotal: 10,
};

const setUp = (props: IOrderCost) => shallow(<OrderCost {...props} />);

describe('OrderCost component', () => {
  let component: any;

  beforeEach(() => {
    component = setUp(props);
  });

  it('should be defined', () => {
    const wrapper = component.find(SummaryCost);
    expect(wrapper).toBeDefined();
  });

  it('should match snapshot', () => {
    const wrapper = renderer.create(<OrderCost {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
