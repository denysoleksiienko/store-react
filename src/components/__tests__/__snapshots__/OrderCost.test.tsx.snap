// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCost component should match snapshot 1`] = `
<div
  className="sc-fFubgz igwYDr"
>
  <div
    className="sc-hHftDr gKZifq"
  >
    <span>
      Subtotal
    </span>
    <span>
      $
      10
    </span>
  </div>
  <div
    className="sc-hHftDr gKZifq"
  >
    <span>
      Shipping
    </span>
    <span>
      Free
    </span>
  </div>
  <div
    className="sc-hHftDr gKZifq"
  >
    <span>
      Taxes
    </span>
    <span>
      $10
    </span>
  </div>
  <hr />
  <div
    className="sc-hHftDr gKZifq"
  >
    <span
      className="sc-dIUggk jiCKLJ"
    >
      Total
    </span>
    <span
      className="sc-dIUggk jiCKLJ"
    >
      $
      20
    </span>
  </div>
</div>
`;
